include/boost/geometry/algorithms/detail/envelope/multipoint.hpp

```diff
@@ -0,0 +1,362 @@
+             ++it)
+        {
+            coordinate_type lon = geometry::get<0>(*it);
+            coordinate_type lat = geometry::get<1>(*it);
+
+            math::normalize_spheroidal_coordinates
+                <
+                    typename coordinate_system<MultiPoint>::type::units,
+                    coordinate_type
+                >(lon, lat);
+
+            if (math::equals(lat, Constants::min_latitude()))
+            {
+                has_south_pole = true;
+            }
+            else if (math::equals(lat, Constants::max_latitude()))
+            {
+                has_north_pole = true;
+            }
+            else
+            {
+                *oit++ = lon_lat_pair<coordinate_type>(lon, lat);
+            }
+        }
+    }
+
+    template <typename SortedRange, typename Value>
+    static inline Value maximum_gap(SortedRange const& sorted_range,
+                                    Value& max_gap_left,
+                                    Value& max_gap_right)
+    {
+        typedef typename boost::range_iterator
+            <
+                SortedRange const
+            >::type iterator_type;
+
+        iterator_type it1 = boost::begin(sorted_range), it2 = it1;
+        ++it2;
+        max_gap_left = it1->first;
+        max_gap_right = it2->first;
+
+        Value max_gap = max_gap_right - max_gap_left;
+        for (++it1, ++it2; it2 != boost::end(sorted_range); ++it1, ++it2)
+        {
+            Value gap = it2->first - it1->first;
+            if (math::larger(gap, max_gap))
+            {
+                max_gap_left = it1->first;
+                max_gap_right = it2->first;
+                max_gap = gap;
+            }
+        }
+
+        return max_gap;
+    }
+
+    template
+    <
+        typename Constants,
+        typename CoordinateRange,
+        typename LongitudeLess,
+        typename CoordinateType
+    >
+    static inline void get_min_max_longitudes(CoordinateRange& range,
+                                              LongitudeLess const& lon_less,
+                                              CoordinateType& lon_min,
+                                              CoordinateType& lon_max)
+    {
+        typedef typename boost::range_iterator
+            <
+                CoordinateRange const
+            >::type iterator_type;
+
+        // compute min and max longitude values
+        std::pair<iterator_type, iterator_type> min_max_longitudes
+            = boost::minmax_element(boost::begin(range),
+                                    boost::end(range),
+                                    lon_less);
+
+        lon_min = min_max_longitudes.first->first;
+        lon_max = min_max_longitudes.second->first;
+
+        // if the longitude span is "large" compute the true maximum gap
+        if (math::larger(lon_max - lon_min, Constants::half_period()))
+        {
+            std::sort(boost::begin(range), boost::end(range), lon_less);
```

Comment:
Is your question really about sorting, or whether we can do it in linear time as opposed to linearithmic time?

I will try to answer both:

1. Yes we can do away without sorting. For example, we can use the sweeping framework I have developed as part of this PR. Instead of sorting all longitudes, I can push them in a priority queue and successively pop them, in order to compute the maximum gap. But I am not really sure we gain something using the sweep approach as opposed to sorting.
2. Yes, it is possible to compute the maximum gap in linear time. There is a bin-based algorithm, that also requires the use of the `floor()` function. You may take a look at [these slides](http://cgm.cs.mcgill.ca/~godfried/teaching/dm-reading-assignments/Maximum-Gap-Problem.pdf), explaining this bin-based linear time algorithm. I am not convinced that in practice it would be faster. I can certainly implement it, either now or at a later time and do benchmarks.